src/core/comm/coll.cc

```diff
@@ -0,0 +1,451 @@
+    thread_comms[global_comm->unique_id].buffers = nullptr;
+    free(thread_comms[global_comm->unique_id].displs);
+    thread_comms[global_comm->unique_id].displs = nullptr;
+    __sync_synchronize();
+    thread_comms[global_comm->unique_id].ready_flag = false;
+  }
+  __sync_synchronize();
+  volatile ThreadComm* data = &(thread_comms[global_comm->unique_id]);
+  while (data->ready_flag != false) { data = &(thread_comms[global_comm->unique_id]); }
+#endif
+  global_comm->status = false;
+  return CollSuccess;
+}
+
+int collAlltoallv(const void* sendbuf,
+                  const int sendcounts[],
+                  const int sdispls[],
+                  CollDataType sendtype,
+                  void* recvbuf,
+                  const int recvcounts[],
+                  const int rdispls[],
+                  CollDataType recvtype,
+                  CollComm global_comm)
+{
+  log_coll.print("Alltoallv: global_rank %d, mpi_rank %d, unique_id %d, comm_size %d",
+                 global_comm->global_rank,
+                 global_comm->mpi_rank,
+                 global_comm->unique_id,
+                 global_comm->global_comm_size);
+#if defined(LEGATE_USE_GASNET)
+  return collAlltoallvMPI(
+    sendbuf, sendcounts, sdispls, sendtype, recvbuf, recvcounts, rdispls, recvtype, global_comm);
+#else
+  return collAlltoallvLocal(
+    sendbuf, sendcounts, sdispls, sendtype, recvbuf, recvcounts, rdispls, recvtype, global_comm);
+#endif
+}
+
+int collAlltoall(const void* sendbuf,
+                 int sendcount,
+                 CollDataType sendtype,
+                 void* recvbuf,
+                 int recvcount,
+                 CollDataType recvtype,
+                 CollComm global_comm)
+{
+  log_coll.print("Alltoall: global_rank %d, mpi_rank %d, unique_id %d, comm_size %d",
+                 global_comm->global_rank,
+                 global_comm->mpi_rank,
+                 global_comm->unique_id,
+                 global_comm->global_comm_size);
+#if defined(LEGATE_USE_GASNET)
+  return collAlltoallMPI(sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype, global_comm);
+#else
+  return collAlltoallLocal(sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype, global_comm);
+#endif
+}
+
+int collGather(const void* sendbuf,
+               int sendcount,
+               CollDataType sendtype,
+               void* recvbuf,
+               int recvcount,
+               CollDataType recvtype,
+               int root,
+               CollComm global_comm)
+{
+  log_coll.print("Gather: global_rank %d, mpi_rank %d, unique_id %d, comm_size %d",
+                 global_comm->global_rank,
+                 global_comm->mpi_rank,
+                 global_comm->unique_id,
+                 global_comm->global_comm_size);
+#if defined(LEGATE_USE_GASNET)
+  return collGatherMPI(
+    sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype, root, global_comm);
+#else
+  printf("Not implemented\n");
```

Comment:
should be `fprintf(stderr, "Not implemented\n");`. I feel we shouldn't even add this to the API if we don't support it in all cases.